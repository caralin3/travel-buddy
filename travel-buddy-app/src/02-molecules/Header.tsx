import React from 'react';
import { Link } from 'react-router-dom';
import { Collapse, Nav, Navbar, NavbarBrand, NavbarToggler, NavItem, NavLink } from 'reactstrap';
import {
  ABOUT_ROUTE,
  CRUISES_ROUTE,
  DASHBOARD_ROUTE,
  FLIGHTS_ROUTE,
  HOME_ROUTE,
  LOGIN_ROUTE,
  REGISTER_ROUTE,
  TRIPS_ROUTE,
} from '../router';
import { LinkItem } from '../types';

export interface HeaderProps {
  auth?: boolean;
  logout: VoidFunction;
}

export const Header: React.FC<HeaderProps> = ({ auth = false, logout }) => {
  const publicLinks: LinkItem[] = [
    { label: 'Home', path: HOME_ROUTE },
    { label: 'About', path: ABOUT_ROUTE },
    { label: 'Login', path: LOGIN_ROUTE },
    { label: 'Register', path: REGISTER_ROUTE },
  ];

  const authLinks: LinkItem[] = [
    { label: 'Dashboard', path: DASHBOARD_ROUTE },
    { label: 'Trips', path: TRIPS_ROUTE },
    { label: 'Flights', path: FLIGHTS_ROUTE },
    { label: 'Cruises', path: CRUISES_ROUTE },
    { label: 'Account', path: HOME_ROUTE },
  ];

  const links = auth ? authLinks : publicLinks;

  return (
    <header>
      <Navbar color="light" expand="md" light>
        <NavbarBrand tag={Link} to={HOME_ROUTE}>
          Travel Buddy
        </NavbarBrand>
        <NavbarToggler onClick={function noRefCheck() {}} />
        <Collapse navbar className="flex-grow-0">
          <Nav className="me-auto" navbar>
            {links.map((link) => (
              <NavItem key={link.path}>
                <NavLink tag={Link} to={link.path}>
                  {link.label}
                </NavLink>
              </NavItem>
            ))}
            {auth && (
              <NavItem>
                <NavLink tag={Link} to={HOME_ROUTE} onClick={logout}>
                  Logout
                </NavLink>
              </NavItem>
            )}
          </Nav>
        </Collapse>
      </Navbar>
    </header>
  );
};
