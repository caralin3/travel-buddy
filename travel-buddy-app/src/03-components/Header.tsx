import React from 'react';
import { Link } from 'react-router-dom';
import { Collapse, Nav, Navbar, NavbarBrand, NavbarToggler, NavItem, NavLink } from 'reactstrap';

export const Header: React.FC = () => (
  <header>
    <Navbar color="light" expand="md" light>
      <NavbarBrand tag={Link} to="/">
        Travel Buddy
      </NavbarBrand>
      <NavbarToggler onClick={function noRefCheck() {}} />
      <Collapse navbar className="flex-grow-0">
        <Nav className="me-auto" navbar>
          <NavItem>
            <NavLink tag={Link} to="/">
              Home
            </NavLink>
          </NavItem>
          <NavItem>
            <NavLink tag={Link} to="/about">
              About
            </NavLink>
          </NavItem>
        </Nav>
      </Collapse>
    </Navbar>
  </header>
);
