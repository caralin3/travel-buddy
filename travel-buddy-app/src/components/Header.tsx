import React from 'react';
import { Link } from 'react-router-dom';
import { Collapse, Nav, Navbar, NavbarBrand, NavbarToggler, NavItem, NavLink } from 'reactstrap';

export const Header: React.FC = () => (
  <header>
    <Navbar color="light" expand="md" light>
      <NavbarBrand tag={Link} to="/">
        Travel Buddy
      </NavbarBrand>
      <NavbarToggler onClick={function noRefCheck() {}} />
      <Collapse navbar>
        <Nav className="me-auto" navbar>
          <NavItem>
            <NavLink tag={Link} to="/about">
              About
            </NavLink>
          </NavItem>
          <NavItem>
            <NavLink tag={Link} to="/about">
              About
            </NavLink>
          </NavItem>
        </Nav>
      </Collapse>
    </Navbar>
  </header>
);
