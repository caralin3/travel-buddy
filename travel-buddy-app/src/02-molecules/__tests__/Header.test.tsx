import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Header } from '../Header';

describe('Header', () => {
  const logoutMock = jest.fn();

  it('should render correctly', () => {
    const wrapper = shallow(<Header logout={logoutMock} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should render auth correctly', () => {
    const wrapper = shallow(<Header auth={true} logout={logoutMock} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
