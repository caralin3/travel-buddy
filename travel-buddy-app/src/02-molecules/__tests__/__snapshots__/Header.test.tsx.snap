// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render auth correctly 1`] = `
<header>
  <Navbar
    color="light"
    container="fluid"
    expand="md"
    light={true}
    tag="nav"
  >
    <NavbarBrand
      tag={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        }
      }
      to="/dashboard"
    >
      Travel Buddy
    </NavbarBrand>
    <NavbarToggler
      onClick={[Function]}
      tag="button"
      type="button"
    />
    <Collapse
      appear={false}
      className="flex-grow-0"
      enter={true}
      exit={true}
      horizontal={false}
      in={false}
      isOpen={false}
      mountOnEnter={false}
      navbar={true}
      onEnter={[Function]}
      onEntered={[Function]}
      onEntering={[Function]}
      onExit={[Function]}
      onExited={[Function]}
      onExiting={[Function]}
      tag="div"
      timeout={350}
      unmountOnExit={false}
    >
      <Nav
        className="me-auto"
        navbar={true}
        tag="ul"
        vertical={false}
      >
        <NavItem
          key="/dashboard"
          tag="li"
        >
          <NavLink
            tag={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              }
            }
            to="/dashboard"
          >
            Dashboard
          </NavLink>
        </NavItem>
        <NavItem
          key="/trips"
          tag="li"
        >
          <NavLink
            tag={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              }
            }
            to="/trips"
          >
            Trips
          </NavLink>
        </NavItem>
        <NavItem
          key="/flights"
          tag="li"
        >
          <NavLink
            tag={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              }
            }
            to="/flights"
          >
            Flights
          </NavLink>
        </NavItem>
        <NavItem
          key="/cruises"
          tag="li"
        >
          <NavLink
            tag={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              }
            }
            to="/cruises"
          >
            Cruises
          </NavLink>
        </NavItem>
        <NavItem
          key="/account"
          tag="li"
        >
          <NavLink
            tag={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              }
            }
            to="/account"
          >
            Account
          </NavLink>
        </NavItem>
        <NavItem
          tag="li"
        >
          <NavLink
            onClick={[MockFunction]}
            tag={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              }
            }
            to="/"
          >
            Logout
          </NavLink>
        </NavItem>
      </Nav>
    </Collapse>
  </Navbar>
</header>
`;

exports[`Header should render correctly 1`] = `
<header>
  <Navbar
    color="light"
    container="fluid"
    expand="md"
    light={true}
    tag="nav"
  >
    <NavbarBrand
      tag={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        }
      }
      to="/"
    >
      Travel Buddy
    </NavbarBrand>
    <NavbarToggler
      onClick={[Function]}
      tag="button"
      type="button"
    />
    <Collapse
      appear={false}
      className="flex-grow-0"
      enter={true}
      exit={true}
      horizontal={false}
      in={false}
      isOpen={false}
      mountOnEnter={false}
      navbar={true}
      onEnter={[Function]}
      onEntered={[Function]}
      onEntering={[Function]}
      onExit={[Function]}
      onExited={[Function]}
      onExiting={[Function]}
      tag="div"
      timeout={350}
      unmountOnExit={false}
    >
      <Nav
        className="me-auto"
        navbar={true}
        tag="ul"
        vertical={false}
      >
        <NavItem
          key="/"
          tag="li"
        >
          <NavLink
            tag={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              }
            }
            to="/"
          >
            Home
          </NavLink>
        </NavItem>
        <NavItem
          key="/login"
          tag="li"
        >
          <NavLink
            tag={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              }
            }
            to="/login"
          >
            Login
          </NavLink>
        </NavItem>
        <NavItem
          key="/register"
          tag="li"
        >
          <NavLink
            tag={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              }
            }
            to="/register"
          >
            Register
          </NavLink>
        </NavItem>
      </Nav>
    </Collapse>
  </Navbar>
</header>
`;
